torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -50,6 +50,26 @@ struct TORCH_API AccumulateGrad : public Node {
         // holding references to the Tensor and that is fine since these are not
         // exposed to the user.
         update_grad(new_grad.detach());
+      } else if (
+          !GradMode::is_enabled() && new_grad.is_sparse() &&
+          new_grad._indices().is_contiguous() &&
+          new_grad._values().is_contiguous() &&
+          new_grad._indices().use_count() <= num_expected_refs &&
+          new_grad._values().use_count() <= num_expected_refs &&
+          new_grad.use_count() <= num_expected_refs) {
+        // Can't detach sparse tensor (since metadata changes are not allowed
+        // after detach), so just create a new one for the grad which is a
+        // shallow copy. We need a shallow copy so that modifying the original
+        // grad tensor doesn't modify the grad we accumulate.
+        // We only skip clone if indices and values themselves are contiguous
+        // for backward compatiblity reasons. Since without this optimization,
+        // earlier we would clone the entire SparseTensor which cloned indices
+        // and values.
```

Comment:
ISTR, weren't you going to file a GitHub issue for the `detach()` issue and link it here?